src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateHandler.java

```diff
@@ -68,154 +75,188 @@ public TransactionActivateHandler(
-                            );
-                            return Boolean.TRUE.equals(isValidPaymentToken)
-                                    ? Mono.just(partialPaymentRequestInfo)
-                                            .doOnSuccess(
-                                                    p -> log.info(
-                                                            "PaymentRequestInfo cache hit for {} with valid paymentToken {}",
-                                                            rptId,
-                                                            p.paymentToken()
+                        ).flatMap(
+                                cacheResult -> {
+                                    PaymentNoticeInfoDto paymentNotice = cacheResult.getT1();
+                                    PaymentRequestInfo partialPaymentRequestInfo = cacheResult.getT2();
+                                    Boolean isValidPaymentToken = isValidPaymentToken(
+                                            partialPaymentRequestInfo.paymentToken()
+                                    );
+                                    return Boolean.TRUE.equals(isValidPaymentToken)
+                                            ? Mono.just(partialPaymentRequestInfo)
+                                                    .doOnSuccess(
+                                                            p -> log.info(
+                                                                    "PaymentRequestInfo cache hit for {} with valid paymentToken {}",
+                                                                    p.id(),
+                                                                    p.paymentToken()
+                                                            )
                                                     )
-                                            )
-                                    : nodoOperations
-                                            .activatePaymentRequest(partialPaymentRequestInfo, newTransactionRequestDto)
-                                            .doOnSuccess(
-                                                    p -> log.info(
-                                                            "Nodo activation for {} with paymentToken {}",
-                                                            rptId,
-                                                            p.paymentToken()
+                                            : nodoOperations
+                                                    .activatePaymentRequest(
+                                                            partialPaymentRequestInfo,
+                                                            paymentNotice.getPaymentContextCode(),
+                                                            paymentNotice.getAmount(),
+                                                            multiplePaymentNotices
                                                     )
```

Comment:
Correct me if I'm wrong on this scenario:
* `getPaymentRequestInfoFromCache` doesn't have a cache hit, so only `rptId`, `isNM3` and `paymentToken` fields are initialized (and `paymentToken` is the only one null)
* `isValidPaymentToken` is false because payment token is null
* `nodoOperations.activatePaymentRequest` is callled
* There the method accesses other fields such as `paTaxCode` without modifying them, and are then persisted with null values

If my analysis is correct it's a bug right?

Reply:
to be fixed as a separate bug fixing